### PR TITLE
Unit tests for SSM package

### PR DESCRIPTION
**Pull request by Madhur97**
**What type of PR is this?**
/area testing

**What this PR does / why we need it**:
This PR increases the unit tests coverage for pkg/cloud/services/ssm package.
Coverage increased from 17.1% to 97.6%

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #2910 

**Special notes for your reviewer**:
Please add missing test scenarios [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2910) for tracking, if any.
**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Add unit tests for `pkg/cloud/services/ssm` package
```

### CODE REVIEW

File: pkg/cloud/services/ssm/secret_test.go [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+	}
+	cs, err := scope.NewClusterScope(scope.ClusterScopeParams{
+		Client:     client,
+		Cluster:    cluster,
+		AWSCluster: &infrav1.AWSCluster{},
+	})
+	if err != nil {
```
Comment (shivi28) [inline pkg/cloud/services/ssm/secret_test.go:271]: nit: you can directly return `cs, err` from this function 
```
func getClusterScope(client client.Client) (*scope.ClusterScope, error) {
	cluster := &clusterv1.Cluster{
		ObjectMeta: metav1.ObjectMeta{
			Name: "test",
		},
	}
	return scope.NewClusterScope(scope.ClusterScopeParams{
		Client:     client,
		Cluster:    cluster,
		AWSCluster: &infrav1.AWSCluster{},
	})
}
```

File: pkg/cloud/services/ssm/secret_test.go [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+		return nil, err
+	}
+	return cs, nil
+}
+
+func getMachineScope(client client.Client, clusterScope *scope.ClusterScope) (*scope.MachineScope, error) {
```
Comment (shivi28) [inline pkg/cloud/services/ssm/secret_test.go:277]: nit: Similarly, you can directly return ms, err from this function
```
func getMachineScope(client client.Client, clusterScope *scope.ClusterScope) (*scope.MachineScope, error) {
	return scope.NewMachineScope(scope.MachineScopeParams{
		Client:       client,
		Cluster:      clusterScope.Cluster,
		Machine:      &clusterv1.Machine{},
		InfraCluster: clusterScope,
		AWSMachine: &infrav1.AWSMachine{
			ObjectMeta: metav1.ObjectMeta{
				Name: "infra-cluster",
			},
		},
	})
}
```

File: pkg/cloud/services/ssm/secret_test.go [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/scope"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/ssm/mock_ssmiface"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+func TestService_Create(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	generateBytes := func(count int64) []byte {
```
Comment (Ankitasw) [inline pkg/cloud/services/ssm/secret_test.go:46]: nit : Can we move `generateBytes`, `check`, `sortTagsByKey` to func outside `TestService_Create` for better readability?

File: pkg/cloud/services/ssm/secret_test.go [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+		Cluster:      clusterScope.Cluster,
+		Machine:      &clusterv1.Machine{},
+		InfraCluster: clusterScope,
+		AWSMachine: &infrav1.AWSMachine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "infra-cluster",
+			},
+		},
+	})
+	if err != nil {
```
Comment (Ankitasw) [inline pkg/cloud/services/ssm/secret_test.go:289]: To expect error during our tests, we can use gomega.Expect() here

File: pkg/cloud/services/ssm/secret_test.go (hunk 1 above)
Comment (Ankitasw) [inline pkg/cloud/services/ssm/secret_test.go:271]: To expect error during our tests, we can use gomega.Expect() here

File: pkg/cloud/services/ssm/secret_test.go (hunk 3 above)
Comment (Madhur97) [reply pkg/cloud/services/ssm/secret_test.go:46]: These methods are relevant to TestService_Create() only, so they are present inside this test. The scope of usage of these methods will remain within this test only.

File: pkg/cloud/services/ssm/secret_test.go (hunk 4 above)
Comment (Madhur97) [reply pkg/cloud/services/ssm/secret_test.go:289]: gomega.Expect() has been used wherever applicable within this file. It has not been used in this method as it is outside of all tests and adding an argument in this method to enforce usage of gomega didn't seem necessary. 
Anyways, as per @shivi28 comment, this piece of code will be removed now.

File: pkg/cloud/services/ssm/secret_test.go (hunk 1 above)
Comment (Madhur97) [reply pkg/cloud/services/ssm/secret_test.go:271]: Replied in other duplicate comment

Comment (Ankitasw) [reply pkg/cloud/services/ssm/secret_test.go]: Replied above

File: pkg/cloud/services/ssm/secret_test.go (hunk 3 above)
Comment (Ankitasw) [reply pkg/cloud/services/ssm/secret_test.go:46]: Since it's a nit, I will leave this to you.

File: pkg/cloud/services/ssm/secret_test.go (hunk 4 above)
Comment (Madhur97) [reply pkg/cloud/services/ssm/secret_test.go]: Oh, now I got it, we are trying to bring consistency in error assertions in other PRs as well by using gomega.
+1 for this as it is more concise/readable. Will start following this :)

**Pull request merged**